### DISCUSSION TITLE
"Heuristic protein inference"

### DISCUSSION THREAD

**Discussion by ibradic95** [vdemichev/DiaNN#680] [Q&A]
Hi,

I went through your explanation of DIA-NN search parameters, but I didn't understand the thing with "heuristic protein inference".

I used DIA-NN for the first time in library-free mode. I tried to have heuristic protein inference off the first time because I understood it is not as accurate for the quantification based on information on the DIA-NN GitHub page. However, I got many proteins that were duplicated and had the same majority protein ID. When I turned the heuristic protein inference on, I got a much nicer pg.matrix without duplciates.
What is the solution to remove duplicated proteins? Should I use some R-package to process the data, or did I use the wrong parameters in the searches which I attached (like "no shared spectra" was off in the first search)?

These were my search parameters.
![image](https://user-images.githubusercontent.com/118803549/235098976-0398daed-514f-4419-81b3-5a0eaa59c611.png)
![image](https://user-images.githubusercontent.com/118803549/235099077-7c983d47-bae9-4a17-bb14-8c4da5fbbf79.png)

Thank you and best regards,
Ivan

**Comment by vdemichev** [comment]
Hi Ivan,

Heuristic protein inference is the easiest way to analyse things for most experiments, that's why it's on by default. Just need to always keep in mind that maximum-parsimony protein grouping might not be perfect, hence need to be cautious in making conclusions about individual proteins, unless identified and quantified using proteotypic peptides. Yes, turning it off will result in 'duplicate' proteins.

'No shared spectra' is a very important algorithm and I suggest to always keep it enabled.
NNs - better to use single-pass mode.

Best,
Vadim

**Comment by barny116** [comment]
Hi Vadim, 
I have a question about this post from a while ago. 
I am not understanding, which setting gets manipulated to ensure identification using proteotypic peptides? 
Thanks!

**Reply by vdemichev** [reply]
Hi Barny,

If you filter the main report using Protein.Q.Value, this ensures the identifications are carried out using proteotypic peptides.

Best,
Vadim

**Comment by bluemooninvestor** [comment]
But why would there be duplicate proteins if the "No shared spectra" is turned on (while not using heuristic protein inference)? No shared spectra means no shared peptides, right? There should be no duplicates in that case.

Also, the application says that heuristic protein inference is not recommended for differential expression analysis. Usually differential expression analysis is the main purpose.

**Reply by vdemichev** [reply]
"No shared spectra" is a very different algorithm, it refers to what is described is 'removal of interfering precursors' in the Nature Methods  DIA-NN paper.

> Also, the application says that heuristic protein inference is not recommended for differential expression analysis. Usually differential expression analysis is the main purpose.

This should be no longer the case in DIA-NN 1.9, i.e. it no longer shows this warning. Currently, whether it's a good thing or not, maximum parsimony algorithms like DIA-NN's heuristic inference are the gold standard in the field.

**Reply by bluemooninvestor** [reply]
Does it just no longer show the warning, or the algorithm has undergone some change to make it compatible with DEA? Sorry it wasn't clear.

**Reply by bluemooninvestor** [reply]
And could you please tell what I should do to have output of unique proteins quantified using unique peptides only?

I guess protein inference should be "genes", and "heuristic inference" to be turned off, anything else? I want accurate DEA, hence asking.

**Reply by bluemooninvestor** [reply]
Can you please clarify

Thnks

**Reply by vdemichev** [reply]
> Does it just no longer show the warning, or the algorithm has undergone some change to make it compatible with DEA? Sorry it wasn't clear.

This algorithm is not related to protein inference or DEA

> And could you please tell what I should do to have output of unique proteins quantified using unique peptides only?

Please use Genes.MaxLFQ.Unique or do some scripting in R as described here https://github.com/vdemichev/DiaNN/discussions/1115

>  and "heuristic inference" to be turned off

No need to turn off

> I want accurate DEA
- In most cases for most biological questions 'on the systems level' you get better answers with default settings
- If you want to be confident about your specific protein of interest, then (i) yes, maybe use only proteotypic peptides (ii) but better examine (using R) how each peptide of this protein varies between conditions

**Reply by bluemooninvestor** [reply]
Thank you, Vadim.
